src/WGPUCanvas.jl

```diff
@@ -31,22 +31,24 @@ elseif Sys.iswindows()
     include("glfwWindows.jl")
 end
 
+struct ErrorCanvas <: AbstractWGPUCanvas end
 
-function WGPUCore.getCanvas(s::Symbol)
-    if s==:OFFSCREEN
-        return defaultCanvas(OffscreenCanvas)
+function WGPUCore.getCanvas(s::Symbol, size::Tuple{Int, Int} = (500,500))
+    canv = if s==:OFFSCREEN
+        OffscreenCanvas
     elseif s==:GLFW
         if Sys.iswindows()
-            return defaultCanvas(GLFWWinCanvas)
+            GLFWWinCanvas
         elseif Sys.isapple()
-            return defaultCanvas(GLFWMacCanvas)
+            GLFWMacCanvas
         elseif Sys.islinux()
-            return defaultCanvas(GLFWLinuxCanvas)
+            GLFWLinuxCanvas
         end
     else
         @error "Couldn't create canvas"
```

Comment:
wrap it in a function like error("Couln't create canvas"). Then you wouldn't need ErrorCanvas. There is a fallback canvas already.

Reply:
Oh I see your intension. Never mind.

Reply:
Ah, the intention was to always return an `AbstractWGPUCanvas` type
Previously it'd return `nothing` in case of error.

Reply:
Right. I got it. I saw your intention first time itself. It's just that where would you capture error. This error macro is just logging version. Do you see what I mean ? This version is type stable with consistent return value. I like it. But what if we want to capture the error or look at stack trace in terminal? Its a choice we can make to prefer type stability over debuggability. You guys can vote which one is preferred. Thumbs up for just logging.

Reply:
Ok. On second thought this might be better but we will have to capture error in WGPUCore some how by checking for ErrorCanvas return type. 

If I am not thinking straight let me know. I am still learning.

Reply:
Sorry I meant WGPUgfx not WGPUCore.